internal/config/ulimits/ulimits.go

```diff
@@ -23,25 +27,64 @@ func New() *Config {
 	}
 }
 
+// appends a ulimit to the c.Ulimits struct
+func (c *Config) appendUlimit(name string, hard, soft uint64) {
+	// This sucks, but it's the runtime-tools interface
+	c.ulimits = append(c.ulimits, Ulimit{
+		Name: name,
+		Hard: hard,
+		Soft: soft,
+	})
+}
+
+// parses a ulimit from a string config
+func (c *Config) parseUlimit(u string) (name string, hard, soft uint64, err error) {
+	ul, err := units.ParseUlimit(u)
+	if err != nil {
+		return "", 0, 0, fmt.Errorf("unrecognized ulimit %s: %v", u, err)
+	}
+	rl, err := ul.GetRlimit()
+	if err != nil {
+		return "", 0, 0, err
+	}
+	return "RLIMIT_" + strings.ToUpper(ul.Name), rl.Hard, rl.Soft, nil
+}
+
 func (c *Config) LoadUlimits(ulimits []string) error {
 	// Process and initialize ulimits at cri-o start up, so crio fails early if
 	// its misconfigured. After this, we can always refer to config.Ulimits() to get
 	// the configured Ulimits
 	for _, u := range ulimits {
-		ul, err := units.ParseUlimit(u)
+		name, hard, soft, err := c.parseUlimit(u)
 		if err != nil {
-			return fmt.Errorf("unrecognized ulimit %s: %v", u, err)
+			return err
 		}
-		rl, err := ul.GetRlimit()
+		c.appendUlimit(name, hard, soft)
+	}
+	return nil
+}
+
+// Loads ulimits from a profile specified in profilepath
+func (c *Config) LoadUlimitsFromProfile(profilePath string, g *generate.Generator) error {
+	file, err := os.Open(profilePath)
+	if err != nil {
+		return fmt.Errorf("failed to open %s, %v", profilePath, err)
+
+	}
+	defer file.Close()
+
+	scanner := bufio.NewScanner(file)
+	for scanner.Scan() {
+		logrus.Info("setting ulimit: ", scanner.Text())
+		name, hard, soft, err := c.parseUlimit(scanner.Text())
 		if err != nil {
-			return err
+			return fmt.Errorf("Failed to parse ulimit: %v", err)
 		}
-		// This sucks, but it's the runtime-tools interface
-		c.ulimits = append(c.ulimits, Ulimit{
-			Name: "RLIMIT_" + strings.ToUpper(ul.Name),
-			Hard: rl.Hard,
-			Soft: rl.Soft,
-		})
+		g.AddProcessRlimits(name, hard, soft)
+	}
+
+	if err := scanner.Err(); err != nil {
+		return fmt.Errorf("reading failed with error: %v", err)
```

Comment:
maybe it is clearer if we specify what we tried to read: `reading ulimit profile failed...`, and we can use Wrapf here too

Reply:
fixed !